packages/xsnap/src/xsnap.js

```diff
@@ -121,7 +163,25 @@ export function xsnap(options) {
         xsnapProcess.kill();
         throw new Error('xsnap protocol error: received empty message');
       } else if (message[0] === OK) {
-        return message.subarray(1);
+        let compute = NaN;
+        const semi = message.indexOf(SEMI, 1);
+        if (semi >= 0) {
+          // The message is `.938586;`.
```

Comment:
I think I understand the need for the semicolon encoding now, but it feels off. Fairly soon, we're going to be returning multiple values from the child, right? Not just CPU usage, but something about peak/current memory allocation? I suspect we'll need something more flexible than `/\.\d+;/`. I guess the deeper issue is the handoff in `xsnap.c` between the JS supervisor code (stashing some "how did it go" value, a string I guess) and the C code that needs to wrap that in a netstring and send it back over the pipe. On one hand, I want the supervisor code to get the metering values, so it can include them next to any status data before it `JSON.stringify`s them into the stashed place. On the other hand, I want the meter to be sampled *after* all JS has finished running, so we aren't looking at meter values while the meter is still in use (which additionally raises the question of what happens if the meter expires after we've sampled it but before we return back into C).

I guess I'm avoiding something: cranks failing for metering reasons is very different than cranks failing for e.g. illegal syscalls. The (JS) supervisor can discover (and report) the latter. But a metering failure will exit the entire JS engine, including the supervisor, so both a metering failure and reporting the meter remaining are things that the C code must do, which means they're necessarily delivered in a different chunk of data than whatever the supervisor provides. Bummer.

Ok so maybe the generalized answer is for each parent-to-child message to result in *two* netstrings in response, glued together. The first can be what the C code reports ("success: no metering violation, here's the remaining meter, or failure because of meter underflow), and the second can be what the JS supervisor code reports ("success: no illegal syscalls"). If the C code sees an error, the second value may be empty (perhaps we synthesize an empty object, just `2:{},`.

Reply:
> Ok so maybe the generalized answer is for each parent-to-child message to result in two netstrings in response, glued together.

I had implemented something like that in the first version of this PR (with the `'M'` command code to report metering results), but it was shot down.  Do you want me to resurrect it?

Reply:
> I suspect we'll need something more flexible than `/\.\d+;/`

There's nothing preventing the evolution of the coupling between `xsnap.c` and `xsnap.js`.  I half-considered `/\.[^\0]*\0/` to encode arbitrary JSON, but opted for the minimal starting point.